File: backend/app/schemas/producto.py
from sqlmodel import SQLModel
from typing import Optional, List

class ProductoCreate(SQLModel):
    nombre: str
    precio: float
    descripcion: Optional[str] = None
    categoria_id: Optional[int] = None
    # Esta es la lista que manda tu React (ej: [1, 3])
    ingrediente_ids: List[int] = []

class ProductoRead(SQLModel):
    id: int
    nombre: str
    precio: float
    descripcion: Optional[str] = None
    categoria_id: Optional[int] = None

File: frontend/src/types/producto.ts
export interface Producto { 
    id: number; 
    nombre: string; 
    precio: number; 
    descripcion?: string; 
    categoria_id?: number; 
}
export interface ProductoCreate { 
    nombre: string; 
    precio: number; 
    descripcion?: string; 
    categoria_id?: number; 
    ingrediente_ids: number[]; 
}